### PR TITLE
network/room (server): add message/join/leave/kick/ban/unban/game name logging

### PR DESCRIPTION
**Pull request by vvanelslande**
This allows you to see what happens in your rooms without joining them if you use citra-room.

Requested by `Drgn#6630` in the MHXX Discord server.

Sample logs:
```
[  10.294862] Network <Info> network\room.cpp:Network::Room::RoomImpl::SendStatusMessage:787: Test (vvanelslande) has joined.
[  12.900549] Network <Info> network\room.cpp:Network::Room::RoomImpl::HandleChatPacket:955: Test (vvanelslande): Test                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
[  17.311502] Network <Info> network\room.cpp:Network::Room::RoomImpl::SendStatusMessage:794: Test (vvanelslande) has left.
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/citra-emu/citra/4987)
<!-- Reviewable:end -->

### CODE REVIEW

**Comment by chris062689** [comment]
> is using the same file for SDL, Qt and dedicated room ok?

I would imagine this logging would only be useful in dedicated-room, but let others chime in as well.

**Comment by vvanelslande** [comment]
> > is using the same file for SDL, Qt and dedicated room ok?
> 
> I would imagine this logging would only be useful in dedicated-room, but let others chime in as well.

I meant "is using the same log file for SDL, Qt and dedicated room ok?"

**Comment by vvanelslande** [comment]
Added log filename option (default: user/log/citra-room.log), requested by `ANTHENA#0474` in the MHXX Discord server.

Added game name logging, requested by `ANTHENA#0474` in the MHXX Discord server.

File: src/dedicated_room/citra-room.cpp [hunk 1]
```diff
@@ -271,6 +289,13 @@ int main(int argc, char** argv) {
         std::cout << "Can not enable Citra Moderators for private rooms\n\n";
     }
 
+    InitializeLogging(log_file);
+
+    Common::DetachedTasks detached_tasks;
+
+    // This is just to be able to link against core
+    gladLoadGL();
```
Comment (zhaowenlan1779) [inline src/dedicated_room/citra-room.cpp]: Why are these two lines moved?

File: src/dedicated_room/citra-room.cpp [hunk 2]
```diff
@@ -162,14 +176,15 @@ int main(int argc, char** argv) {
...
+        {"log-file", required_argument, 0, 'f'},
         {"enable-citra-mods", no_argument, 0, 'e'},
         {"help", no_argument, 0, 'h'},
         {"version", no_argument, 0, 'v'},
         {0, 0, 0, 0},
     };
 
     while (optind < argc) {
-        int arg = getopt_long(argc, argv, "n:d:p:m:w:g:u:t:a:i:hv", long_options, &option_index);
+        int arg = getopt_long(argc, argv, "n:d:p:m:w:g:u:t:a:i:f:hv", long_options, &option_index);
```
Comment (zhaowenlan1779) [inline src/dedicated_room/citra-room.cpp]: I'd prefer something like `-l` for `log-file`. `-f` isn't as clear IMO.

File: src/network/room.cpp [hunk 3]
```diff
@@ -778,6 +778,42 @@ void Room::RoomImpl::SendStatusMessage(StatusMessageTypes type, const std::strin
...
+        } else {
+            LOG_INFO(Network, "{} ({}) has been kicked.", nickname, username);
+        }
+        break;
+    case IdMemberBanned:
+        if (username.empty()) {
+            LOG_INFO(Network, "{} has been banned.", nickname);
+        } else {
+            LOG_INFO(Network, "{} ({}) has been banned.", nickname, username);
+        }
```
Comment (zhaowenlan1779) [inline src/network/room.cpp]: If you are going to use so many `if`s, I think you may as well just do this and use it in every `case`.
```cpp
const std::string display_name = username.empty() ? nickname : fmt::format("{} ({})", nickname, username);
...
```

File: src/network/room.cpp [hunk 4]
```diff
@@ -930,6 +973,21 @@ void Room::RoomImpl::HandleGameNamePacket(const ENetEvent* event) {
...
+                             member->user_data.username);
+                }
+            } else {
+                if (member->user_data.username.empty()) {
+                    LOG_INFO(Network, "{} is playing {}", member->nickname, game_info.name);
+                } else {
+                    LOG_INFO(Network, "{} ({}) is playing {}", member->nickname,
+                             member->user_data.username, game_info.name);
+                }
+            }
```
Comment (zhaowenlan1779) [inline src/network/room.cpp]: ditto. I would prefer having a variable instead of many `if`s

**Pull request merged**